test/gtest/multimarginloss.cpp

```diff
@@ -0,0 +1,113 @@
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include "multimarginloss.hpp"
+#include <miopen/env.hpp>
+
+MIOPEN_DECLARE_ENV_VAR_STR(MIOPEN_TEST_FLOAT_ARG)
+MIOPEN_DECLARE_ENV_VAR_BOOL(MIOPEN_TEST_ALL)
+
+namespace multimarginloss {
+
+std::string GetFloatArg()
+{
+    const auto& tmp = env::value(MIOPEN_TEST_FLOAT_ARG);
+    if(tmp.empty())
+    {
+        return "";
+    }
+    return tmp;
+}
+
+struct MultiMarginLossForwardTestFloat : MultiMarginLossForwardTest<float>
+{
+};
+
+struct MultiMarginLossForwardTestHalf : MultiMarginLossForwardTest<half_float::half>
+{
+};
+
+struct MultiMarginLossForwardTestBFloat16 : MultiMarginLossForwardTest<bfloat16>
+{
+};
+
+} // namespace multimarginloss
+
+using multimarginloss::MultiMarginLossForwardTestBFloat16;
+using multimarginloss::MultiMarginLossForwardTestFloat;
+using multimarginloss::MultiMarginLossForwardTestHalf;
+
+TEST_P(MultiMarginLossForwardTestFloat, MMLFwdTest)
+{
+    if(!MIOPEN_TEST_ALL ||
+       (env::enabled(MIOPEN_TEST_ALL) && env::value(MIOPEN_TEST_FLOAT_ARG) == "--float"))
+    {
+        RunTest();
+        Verify();
+    }
+    else
+    {
+        GTEST_SKIP();
+    }
+};
+
+TEST_P(MultiMarginLossForwardTestHalf, MMLFwdTest)
+{
+    if(!MIOPEN_TEST_ALL ||
+       (env::enabled(MIOPEN_TEST_ALL) && env::value(MIOPEN_TEST_FLOAT_ARG) == "--half"))
+    {
+        RunTest();
+        Verify();
+    }
+    else
+    {
+        GTEST_SKIP();
+    }
+};
+
+TEST_P(MultiMarginLossForwardTestBFloat16, MMLFwdTest)
+{
+    if(!MIOPEN_TEST_ALL ||
+       (env::enabled(MIOPEN_TEST_ALL) && env::value(MIOPEN_TEST_FLOAT_ARG) == "--bfloat16"))
+    {
+        RunTest();
+        Verify();
+    }
+    else
+    {
+        GTEST_SKIP();
+    }
+};
+
+INSTANTIATE_TEST_SUITE_P(MultiMarginLossTestSet,
+                         MultiMarginLossForwardTestFloat,
+                         testing::ValuesIn(MultiMarginLossTestConfigs()));
+INSTANTIATE_TEST_SUITE_P(MultiMarginLossTestSet,
+                         MultiMarginLossForwardTestHalf,
+                         testing::ValuesIn(MultiMarginLossFp16TestConfigs()));
+INSTANTIATE_TEST_SUITE_P(MultiMarginLossTestSet,
+                         MultiMarginLossForwardTestBFloat16,
+                         testing::ValuesIn(MultiMarginLossTestConfigs()));
```

Comment:
May I ask you to rename the tests according to the new naming scheme described [on the wiki](https://github.com/ROCm/MIOpen/wiki/GTest-development)?
Also check this document and properly adjust test assertion. Some of `EXCEPT_` assertions should be replaced with `ASSERT_`. Some of the assertion conditions should be properly used, for example `EXCEPT_TRUE(a == b);` should be replaced with `EXCEPT_EQ(a, b);`.

@seungmanhan @long10024070 @BuiChiTrung @et16kr could you also take into account this wiki page?

Reply:
@BuiChiTrung I've also noticed that, but let's ignore this warning for now.
We've already started that renaming and have already done some work.
I'll update the scheme and rename the test a bit later, once we've got regexp checker for the test names.

Reply:
fixed it